nix-shell does not provide dependencies for building the stdlib refman

The `doc/README.md` claims that `nix-shell` provides all the required dependencies (except for LaTeX packages), but when running `nix-shell` then `make refman-html`, I get an error related to the absence of the `sphinx-build` dependency.

The precise command should be `nix-shell --argstr job stdlib-refman-html` but when I type `make refman-html` in it I get
```
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8): No such file or directory
Done: 99% (2908/2909, 1 left) (jobs: 1)Traceback (most recent call last):
  File "/nix/store/dmc5kancg6klj6fwa4afm3hr0y1as1m9-python3.11-sphinx-7.4.7/bin/.sphinx-build-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/1di8gvk34k7xxajh43yhyzy81k1gh993-python3-3.11.11-env/lib/python3.11/site-packages/sphinx/cmd/build.py", line 369, in main
    locale.setlocale(locale.LC_ALL, '')
  File "/nix/store/0flj08i381bfdfbrly8sk6vs36lfrhnb-python3-3.11.11/lib/python3.11/locale.py", line 627, in setlocale
    return _setlocale(category, locale)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
locale.Error: unsupported locale setting
File "doc/dune", lines 21-36, characters 0-459:
21 | (rule
22 |  (targets
23 |   (dir refman-html))
....
34 |   (env_var SPHINXWARNOPT))
35 |  (action
36 |   (run env sphinx-build -q %{env:SPHINXWARNOPT=-W} -b html sphinx %{targets})))
Command exited with code 1.
make: *** [Makefile:14: refman-html] Error 1
```

Whereas nix-build in the CI works?

--- a/doc/README.md
+++ b/doc/README.md
@@ -44,8 +44,10 @@ with `apt install python3-pip python3-setuptools` on Debian / Ubuntu) then run:
                  antlr4-python3-runtime==4.7.1 pexpect sphinxcontrib-bibtex
 
 Nix users should get the correct development environment to build the
-HTML documentation from Rocq's [`default.nix`](../default.nix) (note this
-doesn't include the LaTeX packages needed to build the full documentation).
+HTML documentation from Rocq's [`default.nix`](../default.nix) (more
+precisely `nix-shell --argstr job stdlib-refman-html`, note this
+doesn't include the LaTeX packages needed to build the full
+documentation).
 
 You can check the dependencies using the `doc/tools/coqrst/checkdeps.py` script.
 

